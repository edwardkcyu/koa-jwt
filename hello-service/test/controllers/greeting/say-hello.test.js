require('dotenv').config({ path: 'test/.env.test' });

const request = require('supertest');

jest.mock('axios');
const axios = require('axios');

const app = require('../../../src/app');
const { HTTP_STATUS } = require('../../../src/lib/constants');

describe('root route', () => {
  beforeEach(() => {
    axios.mockResolvedValue({ data: { userName: 'curry' } });
  });

  afterAll(() => {
    app.server.shutdown(function() {
      console.log('Everything is cleanly shutdown.');
    });
  });

  test('retrieves user name', async () => {
    const response = await request(app.callback())
      .get('/hello')
      .set('Content-Type', 'application/json')
      .set('Authorization', 'Bearer DUMMY_TOKEN');

    expect(response.status).toEqual(HTTP_STATUS.OK);
    expect(response.type).toEqual('application/json');
    expect(response.body).toMatchSnapshot();
  });

  test('error when missing authorization header', async () => {
    const response = await request(app.callback())
      .get('/hello')
      .set('Content-Type', 'application/json');

    expect(response.status).toEqual(HTTP_STATUS.UNAUTHORIZED);
    expect(response.type).toEqual('application/json');
    expect(response.body).toMatchSnapshot();
  });
});
