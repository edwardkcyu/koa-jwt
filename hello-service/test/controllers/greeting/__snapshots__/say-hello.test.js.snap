// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /hello error when missing authorization header 1`] = `
Object {
  "error": Object {
    "code": "E001",
    "message": "Authentication required!",
  },
}
`;

exports[`GET /hello retrieves user name 1`] = `
Object {
  "message": "Hello curry",
}
`;
